### PR TITLE
Many smaller changes

### PR DESCRIPTION
**Pull request by dennisshushack**
1) Can now edit password in User
#55
2) Can now Report Items -> If 3 Reports Item and Pictures of that Item get Deleted #46
3) You can now get a filtered list of all items based on tags -> You only receive those items with the specified tags, excluding yours

4)When deleting a picture the picture counter decreases -> Max 5 pictures per Item

#22 Was allready in there
#13, #20 Now possible to properly edit the title and description

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs21/service/ItemService.java [hunk 1]
```diff
@@ -24,46 +23,51 @@
...
             return item;
         }
     }
+
     //Update the item
-    public void updateItem(Item currentItem,Item userInput){
+    public void updateItem(Item currentItem, Item userInput) {
         // Changes the Description of the item
-        if(userInput.getDescription()!=null){
+        if (userInput.getDescription() != "") {
```
Comment (otopalak) [inline src/main/java/ch/uzh/ifi/hase/soprafs21/service/ItemService.java]: if( !userInput.getDescription.isBlank() ) {...

https://howtodoinjava.com/java11/check-blank-string/

File: src/main/java/ch/uzh/ifi/hase/soprafs21/service/UserService.java [hunk 2]
```diff
@@ -118,6 +118,12 @@ public User updateUser(User currentUser, User userInput){
         if(userInput.getBirthday()!=null){
             currentUser.setBirthday(userInput.getBirthday());
         }
+        // Setting a new Password
+        if(userInput.getPassword()==""){
```
Comment (otopalak) [inline src/main/java/ch/uzh/ifi/hase/soprafs21/service/UserService.java]: .isBlank()

File: src/main/java/ch/uzh/ifi/hase/soprafs21/service/ItemService.java (hunk 1 above)
Comment (dennisshushack) [reply src/main/java/ch/uzh/ifi/hase/soprafs21/service/ItemService.java]: hanni veräbndert

File: src/main/java/ch/uzh/ifi/hase/soprafs21/service/UserService.java (hunk 2 above)
Comment (dennisshushack) [reply src/main/java/ch/uzh/ifi/hase/soprafs21/service/UserService.java]: hanni veränderet

**Review by otopalak — APPROVED** [review]
looks good

**Pull request merged**